src/linear_algebra/span.lean

```diff
@@ -362,11 +374,11 @@ lemma disjoint_span_singleton {K E : Type*} [division_ring K] [add_comm_group E]
 begin
   refine disjoint_def.trans ⟨λ H hx, H x hx $ subset_span $ mem_singleton x, _⟩,
   assume H y hy hyx,
-  obtain ⟨c, hc⟩ := mem_span_singleton.1 hyx,
-  subst y,
-  classical, by_cases hc : c = 0, by simp only [hc, zero_smul],
-  rw [s.smul_mem_iff hc] at hy,
-  rw [H hy, smul_zero]
+  obtain ⟨c, rfl⟩ := mem_span_singleton.1 hyx,
+  by_cases hc : c = 0,
+  { simp only [hc, zero_smul] },
+  { rw [s.smul_mem_iff hc] at hy,
+    rw [H hy, smul_zero] }
```

Comment:
Can shorten this a bit:
```
  { rw [hc, zero_smul] },
  { rw [H ((s.smul_mem_iff hc).mp hy), smul_zero] }
```

Reply:
That feels overly golfed to me, usually we don't put messy term-mode proofs inside `rw`.

The only reason I touched this proof in the first place is because it broke our "use `{` for subgoals" rule.

Reply:
I took the first `rw` change though, thanks!